app/src/test/java/com/example/candyshop/data/CandyItemsRepositoryImplementationTest.kt

```diff
@@ -0,0 +1,241 @@
+import com.example.candyshop.data.model.Candy
+import com.example.candyshop.rules.TestDispatcherRule
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit4.MockKRule
+import io.mockk.unmockkAll
+import kotlinx.coroutines.test.runTest
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.ResponseBody
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+
+class CandyItemsRepositoryImplementationTest {
+    private val result = listOf(
+        Candy(
+            name = "FirstDessert",
+            imageUrl = "url1",
+            id = "12345",
+            price = 10
+        ), Candy(
+            name = "SecondDessert",
+            imageUrl = "url2",
+            id = "54321",
+            price = 10
+        )
+    )
+    private val fakeLocalCandyList = listOf(
+        CandyEntity(
+            name = "FirstDessert",
+            imageUrl = "url1",
+            id = "12345",
+            price = 10
+        ), CandyEntity(
+            name = "SecondDessert",
+            imageUrl = "url2",
+            id = "54321",
+            price = 10
+        )
+    )
+    private val fakeRemoteCandyList = CandyListDto(meals = listOf (
+        CandyDto(
+            name = "FirstDessert",
+            imageUrl = "url1",
+            id = "12345",
+            price = 10
+        ), CandyDto(
+            name = "SecondDessert",
+            imageUrl = "url2",
+            id = "54321",
+            price = 10
+        )
+    ))
+
+    @get:Rule
+    val testDispatcher = TestDispatcherRule()
+
+    @get:Rule
+    val mockkRule = MockKRule(this)
+
+    @RelaxedMockK
+    private lateinit var candyApiService: CandyApiService
+
+    @RelaxedMockK
+    private lateinit var candyDatabase: CandyDatabase
+
+    private lateinit var candyItemsRepositoryImplementation: CandyItemsRepositoryImplementation
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun candyItemsRepositoryImplementation_getCandyItems_loadLocallySuccess() = runTest {
+        coEvery { candyDatabase.candyDao.getAllCandy() } returns fakeLocalCandyList
+        candyItemsRepositoryImplementation = CandyItemsRepositoryImplementation(candyApiService, candyDatabase)
+        val flow = candyItemsRepositoryImplementation.getCandyItems(forceFetchFromRemote = false)
+        flow.collect { resource: Resource<List<Candy>> ->
+            assertEquals(result, resource.data)
+        }
+        coVerify {
+            candyApiService
+            candyDatabase
+            candyDatabase.candyDao.getAllCandy()
```

Comment:
not sure why you are adding this chain: service / database / function call to verify? for me its enough to put here: coVerify { candyDatabase.candyDao.getAllCandy() } because you just want to check if this function was called, but to be honest here you anyway have to provide some response for this function to make it work so this verify is not needed in my opinion because if you remove this bit: coEvery { candyDatabase.candyDao.getAllCandy() } returns fakeLocalCandyList it will be not working anyway